floater/test/test_generators.py

```diff
@@ -186,6 +187,39 @@ def test_pickling_with_land(fs_with_land, tmpdir):
                 assert np.all(fs.__dict__[key][sub_key] == fs_from_pickle.__dict__[key][sub_key])
 
 
+def test_npart_to_2D_array():
+    lon = np.linspace(0, 8, 9, dtype=np.float32)
+    lat = np.linspace(-4, 4, 9, dtype=np.float32)
+    land_mask = np.zeros(81)
+    for i in range(81):
+        if (i%9==0) or (i%9==1):
+            land_mask[i] = 0.
+        else:
+            land_mask[i] = 1.
+    land_mask = land_mask==1.
+    land_mask.shape = (len(lat), len(lon))
+    model_grid = {'lon': lon, 'lat': lat, 'land_mask': land_mask}
+    fs = gen.FloatSet(xlim=(0, 9), ylim=(-4, 5), dx=1, dy=1, model_grid=model_grid)
+    fs.get_rectmesh()
```

Comment:
`fs.get_rectmech()` here is for the initiation of `fs.ocean_bools`. This `ocean_bools` variable cannot be automatically created before calling `subset_floats_from_mask`.